src/acceptance-test/java/org/zalando/nakadi/webservice/timelines/SubscriptionConsumptionTest.java

```diff
@@ -0,0 +1,246 @@
+                        "001-0005-000000000000000000",
+                        "001-0005-000000000000000001"
+                },
+                inTimelineCursors.get()
+        );
+
+        // Will create subscription clone
+        final Subscription clone = createSubscription(
+                RandomSubscriptionBuilder.builder().withEventType(eventType.getName())
+                        .withStartFrom(SubscriptionBase.InitialPosition.BEGIN).build());
+        final CountDownLatch finished2 = new CountDownLatch(1);
+        createParallelConsumer(clone, 5, finished2, inTimelineCursors::set);
+        finished2.await();
+        Assert.assertArrayEquals(
+                new String[]{
+                        "001-0001-000000000000000000",
+                        "001-0001-000000000000000001",
+                        "001-0003-000000000000000000",
+                        "001-0005-000000000000000000",
+                        "001-0005-000000000000000001"
+                },
+                inTimelineCursors.get()
+        );
+    }
+
+    @Test
+    public void test2TimelinesInaRowNoBegin() throws IOException, InterruptedException {
+        final EventType eventType = createEventType();
+        final Subscription subscription = createSubscription(
+                RandomSubscriptionBuilder.builder().withEventType(eventType.getName()).build());
+
+        final CountDownLatch finished = new CountDownLatch(1);
+        final AtomicReference<String[]> inTimelineCursors = new AtomicReference<>();
+        createParallelConsumer(subscription, 2, finished, inTimelineCursors::set);
+        createTimeline(eventType.getName()); // Still old topic
+        createTimeline(eventType.getName()); // New topic
+        createTimeline(eventType.getName()); // Another new topic
+        IntStream.range(0, 2).forEach(idx -> publishEvent(eventType.getName(), "{\"foo\":\"bar\"}"));
+        finished.await();
+        Assert.assertArrayEquals(
+                new String[]{
+                        "001-0003-000000000000000000",
+                        "001-0003-000000000000000001",
+                },
+                inTimelineCursors.get()
+        );
+
+        final Subscription subscription2 = createSubscription(RandomSubscriptionBuilder.builder()
+                .withEventType(eventType.getName()).withStartFrom(SubscriptionBase.InitialPosition.BEGIN).build());
+
+        final CountDownLatch finished2 = new CountDownLatch(1);
+        final AtomicReference<String[]> inTimelineCursors2 = new AtomicReference<>();
+        createParallelConsumer(subscription2, 2, finished2, inTimelineCursors2::set);
+        finished2.await();
+        Assert.assertArrayEquals(
+                new String[]{
+                        "001-0003-000000000000000000",
+                        "001-0003-000000000000000001",
+                },
+                inTimelineCursors2.get()
+        );
+    }
+
+    @Test
+    public void testInTimeCursorsCorrect() {
```

Comment:
This is the consequence of attempt to make it a little bit faster. The initialization of this collection is made in `@BeforeClass` method, and several checks are done in different tests.